dspace-spring-rest/src/main/java/org/dspace/app/rest/link/relation/RelationshipHalLinkFactory.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.link.relation;
+
+import java.util.LinkedList;
+
+import org.atteo.evo.inflector.English;
+import org.dspace.app.rest.RestResourceController;
+import org.dspace.app.rest.link.HalLinkFactory;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.model.hateoas.RelationshipResource;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.Link;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RelationshipHalLinkFactory extends HalLinkFactory<RelationshipResource, RestResourceController> {
```

Comment:
This new class needs JavaDocs to describe it.

Reply:
The JavaDoc has been included